README.md

```diff
@@ -1,13 +1,408 @@
+Download
+--------
 
-# CPP Collections
+[C++ Collections][dl] (18 KB) -- 10 April 2015
 
-[![Build Status](https://travis-ci.org/natebrennand/cpp_collections.svg?branch=master)](https://travis-ci.org/natebrennand/cpp_collections)
+[dl]: https://github.com/natebrennand/cpp_collections/archive/master.zip
 
-Designing a functional collections library for C++.
 
------------------------
+----
+## Introduction
 
+Collections are fundamental to most programming tasks. More importantly, they let you group and process large sets of data. Whether you  want to create a collection of a financial transactions you want to operate on, or you want to calculate something as summing a list of squares between 0 and 10.
 
-See testing and development guidelines in the `src/` directory.
+![Image of Kesiev](http://i.imgur.com/GnCHqIm.png?1) 
 
+Currently, processing collections in C++ is not only far from perfect, but is syntax-wise far behind many current functional programming paradigms.
 
+Languages are used as tools to communicate, and incidentally, the structure and limitations of a C++ determine the way we think. Learning how to think and within the context of a language provides one the ability to be a proper user, well at the same time 
+
+> In 24 hours you might be able to learn some of the syntax of C++ (if you already know another language), but you couldn't learn much about how to use the language. In short, if you were, say, a Basic programmer, you could learn to write programs in the style of Basic using C++ syntax, but you couldn't learn what C++ is actually good (and bad) for. So what's the point? Alan Perlis once said: "A language that doesn't affect the way you think about programming, is not worth knowing" -- [Peter Norvig][1]
+
+[1]:http://www.norvig.com/21-days.html
+
+This is the principle of linguistic relativity also known as the [Sapir-Whorf hypothesis][2] shows that the constraints of a language are what most often limit our thought processing. This idea, when applied to programming languages shows how Lisp programmers use macros for everything, Python programmers even in Ruby have code that looks like python.  
+
+[2]:http://en.wikipedia.org/wiki/Linguistic_relativity
+
+One of the key components to developing C++ is learnability or more importantly, how easy is it for users to accomplish basic tasks. 
+
+C++ Collections is a C++ 11 library that provides lazy evaluation and functional transformations of the data to lighten the burden of using functional operations such as map, range, and filter for the user. To use C++ Streams, download them from Github and simply *#include collections.h* in the header of your file after dragging it in.
+
+----
+## What is C++ Collections?
+C++ Collections provides abstractions on a set of ordered data. All examples below assume that we are `using namespace cpp_collections`.
```

Comment:
I think the second sentence should be moved to a footnote or something along those lines.
Just leave a comment to make it a footnote and we can do that when we transition to latex.